### ISSUE TITLE
Incorrect error issued for return value type.

### ISSUE THREAD

**Issue by Techget** [opis/closure#17]
Hello, I meet some weired error when I use `opis closure` to serialize&unserialize [pimple container](https://pimple.symfony.com/), after unserializing, it seems causing some ambiguity to the return value type.
1. Currently, it can't determine the return type correctly after unserializing, for example:
```
function (Container $container) use ($class, $args, $stacktraces): FormatterInterface {
    $formatter = newv($class, $args); // newv is a function to create new class
    if ($stacktraces !== null) {
        $formatter->includeStacktraces($stacktraces);
    }
    return $formatter;
}
```
In above closure, the formatter is going to be a class extends from `basicFormatter`, and `basicFormatter` implements the `FormatterInterface`, so the return value will be an instance of `FormatterInterface` however I get error message as following:
```
Fatal error: Uncaught TypeError: Return value of Opis\Closure\SerializableClosure::{closure}() must be an instance of FormatterInterface, instance of JsonFormatter returned in closure://function (\Pimple\Container $container) use ($class, $args, $stacktraces): FormatterInterface {
            $formatter = newv($class, $args);
            if ($stacktraces !== null) {
                $formatter->includeStacktraces($stacktraces);
            }
            return $formatter;
        }:
```
2. It does not compatible with alias in use statement, for example:
```
use Aws\S3\S3Client as AwsS3Client;
$container['AwsS3Client'] = function (Container $container) use ($config): AwsS3Client {
      return $container['AwsSdk']->createS3($config);
};
```
Will get error message as `message":"Return value of Opis\\Closure\\SerializableClosure::{closure}() must be an instance of AwsS3Client, instance of Aws\\S3\\S3Client returned","code":500,`

Please help when you get time. Thanks in advance.

**Comment by msarca** [comment]
Actually, Opis Closure is returning the correct type, `instance of Aws\\S3\\S3Client returned`, but you expect an instance of `AwsS3Client`, which is wrong. Try using something like below

```php
use Aws\S3\S3Client as AwsS3Client;
$container[AwsS3Client::class] = function (Container $container) use ($config): AwsS3Client {
      return $container['AwsSdk']->createS3($config);
};
```

**Comment by Techget** [comment]
Thanks for your response, I tried your solution, doesn't work, I think there is some misunderstanding,  `Instance of Aws\\S3\\S3Client returned` is correct but `AwsS3Client` returned should also be correct, since as you can see in above use case, `AwsS3Client` is an alias of `Aws\\S3\\S3Client`. 

And how about the first problem, it can't determine the type correctly, let me give you an simplified example:
```
class A implements interfaceA {}
class B extends A {}
$closure = function(): interfaceA {
   return new B();   // after unserializing, it complains that this closure doesn't return an instance of interface A
}
```
Could you please also have a look on this problem, this really bites me.

**Comment by msarca** [comment]
Sorry, but I can't reproduce this.It will be much more easy if you can provide me a full example, not just pieces of code. The most simple and reproducible example you can provide. Also, please specify which version of Opis Closure and Pimple are you using. Thanks.

**Comment by msarca** [comment]
Finally, I was able to reproduce the error. I will provide a fix asap.

**Comment by msarca** [comment]
This should be fixed in v3.0.9
